examples/sv3d/models/instantmesh/models/geometry/rep_3d/flexicubes.py

```diff
@@ -0,0 +1,624 @@
+        unique_edges, _idx_map, counts = mint.unique(
+            all_edges.to(ms.bfloat16), dim=0, return_inverse=True, return_counts=True
+        )
+        unique_edges = unique_edges.long()
+        mask_edges = occ_n.to(ms.uint8)[unique_edges.reshape(-1)].reshape(-1, 2).sum(-1) == 1
+
+        surf_edges_mask = mask_edges.to(ms.uint8)[_idx_map]
+        counts = counts[_idx_map]
+
+        mapping = ops.ones((unique_edges.shape[0]), dtype=ms.int64) * -1
+        mapping[mask_edges] = ops.arange(mask_edges.sum())
+        # Shaped as [number of cubes x 12 edges per cube]. This is later used to map a cube edge to the unique index
+        # for a surface-intersecting edge. Non-surface-intersecting edges are marked with -1.
+        idx_map = mapping[_idx_map]
+        surf_edges = unique_edges[mask_edges]
+        return surf_edges, idx_map, counts, surf_edges_mask
+
+    def _identify_surf_cubes(self, s_n, cube_fx8):
+        """
+        Identifies grid cubes that intersect with the underlying surface by checking if the signs at
+        all corners are not identical.
+        """
+        occ_n = ms.Tensor(
+            s_n < 0, dtype=ms.uint8
+        )  # bool type cannot be sampled in the following line, needs to be ms tensor
+        occ_fx8 = occ_n[cube_fx8.reshape(-1)].reshape(-1, 8)
+        _occ_sum = ops.sum(occ_fx8, -1)  # 8 verts per cube
+        surf_cubes = (_occ_sum > 0).to(ms.uint8) & (_occ_sum < 8).to(ms.uint8)
+        return surf_cubes, occ_fx8
+
+    def _linear_interp(self, edges_weight, edges_x):
+        """
+        Computes the location of zero-crossings on 'edges_x' using linear interpolation with 'edges_weight'.
+        """
+        edge_dim = edges_weight.dim() - 2
+        assert edges_weight.shape[edge_dim] == 2
+        edges_weight = ops.cat(
+            [
+                mint.index_select(input=edges_weight, index=ms.Tensor(1), dim=edge_dim),
+                -mint.index_select(input=edges_weight, index=ms.Tensor(0), dim=edge_dim),
+            ],
+            edge_dim,
+        )
+        denominator = edges_weight.sum(edge_dim)
+        ue = (edges_x * edges_weight).sum(edge_dim) / denominator
+        return ue
+
+    def _compute_vd(
+        self, x_nx3, surf_cubes_fx8, surf_edges, s_n, case_ids, beta_fx12, alpha_fx8, gamma_f, idx_map, grad_func
+    ):
+        """
+        Computes the location of dual vertices as described in Section 4.2
+        """
+        alpha_nx12x2 = mint.index_select(input=alpha_fx8, index=self.cube_edges, dim=1).reshape(-1, 12, 2)
+        surf_edges_x = mint.index_select(input=x_nx3, index=surf_edges.reshape(-1), dim=0).reshape(-1, 2, 3)
+        surf_edges_s = mint.index_select(input=s_n, index=surf_edges.reshape(-1), dim=0).reshape(-1, 2, 1)
+        zero_crossing = self._linear_interp(surf_edges_s, surf_edges_x)
+
+        idx_map = idx_map.reshape(-1, 12)
+        num_vd = mint.index_select(input=self.num_vd_table, index=case_ids, dim=0)
```

Comment:
Raised error:
```
Exception has occurred: ValueError
For 'IndexSelect', the dimension of 'index' must be 1, but got 0.
```
Funny thing is that this line can be runned inlinely w/o error during the debugpy with a breakpoint.

Reply:
`index` does not take scalar in MS, unlike torch it has to be `[0]` rather than a scalar `0`.